src/rdposix.h

```diff
@@ -108,12 +126,45 @@
  */
 static RD_INLINE RD_UNUSED
 void rd_usleep (int usec, rd_atomic32_t *terminate) {
+#ifdef __370__
+    int left = usec / 1000000;
+
+    while(usec > 0) {
+        unsigned long long stck_start;
+        unsigned long long stck_end;
+
+        __stckf(&stck_start);
+	/* if sleep return non 0 it was interupted
+	 * the return is teh number of second it had
+	 * _not_ slept
+	 */
+	if(left)
+	{
+	    left = sleep(left);
+	}
+	else
+	{
+	    usleep(usec);
+	}
+	if(terminate && rd_atomic32_get(terminate)) {
+	    break;
+	}
+	__stckf(&stck_end);
+	stck_end -= stck_start;
+	/* a storeclock shifted right by 12 is a number of
+	 * microsecond since 1/1/1900
+	 */
+	stck_end >>= 12;
```

Comment:
I'm not sure I understand the question... the goal is to wait for a 'duration'
and that duration is expressed in micro-seconds... the comment above explain why we need to shift 12 to get that